### PR TITLE
feat INT-89: add dynamic list input

### PR DESCRIPTION
**Pull request by suarezcumare**
> adding the dynamic list input to obtain given json element arrangement with the needed values.

### CODE REVIEW

File: src/selector/input/__tests__/dynamic-list.test.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import { testInput } from '../../../test/helpers/test-input';
+import { testSerialization } from '../../../test/helpers/serialization';
+import { inputSelectorDeserializer } from '../deserializer';
+import { testDescription } from '../../../test/helpers/description';
+
+const ONE = 1;
+const THREE = 3;
+const path = '[{"key":"name","value":"name"}]';
+describe('DynamicListInputSelector', () => {
+  const selector = new DynamicListInputSelector('data', path);
```
Comment (smurua-co) [inline src/selector/input/__tests__/dynamic-list.test.ts]: Why is this declared here and used in the end for `testSerialization`? You can make a test for each validation

File: src/selector/input/__tests__/dynamic-list.test.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+  it('gets the expected first level property', () => {
+    const selector = new DynamicListInputSelector('data', '[{"key":"name","value":"name"}]');
```
Comment (smurua-co) [inline src/selector/input/__tests__/dynamic-list.test.ts]: You can use `path` here

File: src/selector/input/__tests__/dynamic-list.test.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import { DynamicListInputSelector } from '../dynamic-list';
+import { testInput } from '../../../test/helpers/test-input';
+import { testSerialization } from '../../../test/helpers/serialization';
+import { inputSelectorDeserializer } from '../deserializer';
+import { testDescription } from '../../../test/helpers/description';
+
+const ONE = 1;
+const THREE = 3;
+const path = '[{"key":"name","value":"name"}]';
+describe('DynamicListInputSelector', () => {
```
Comment (smurua-co) [inline src/selector/input/__tests__/dynamic-list.test.ts]: There's a test missing for multiple selections: 
example
```javascript
	const path = '[{"key":"name","value":"name"}, {"key":"values.string","value":"value"}]';
```

File: src/selector/input/dynamic-list.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    this.path.forEach(list => notStrictEqual(list.value, ''));
+  }
+
+  static deserialize(obj: DynamicListInputSelectorSerialized) {
+    return new this(obj.from, obj.path);
+  }
+
+  describe(): Description {
+    return {
+      text: `the property '${JSON.stringify(this.path)}' in the '${this.from}' object`
```
Comment (smurua-co) [inline src/selector/input/dynamic-list.ts]: In the descriptions it should say that there are multiple properties, maybe you could format the `path` to make it more clear

File: src/selector/input/dynamic-list.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    };
+  }
+
+  apply(input: Input) {
+    const object = input.get(this.from);
+    return this.path.reduce((curr: {}, obj: DynamicListObject) =>
+    // tslint:disable-next-line: no-parameter-reassignment
```
Comment (warseph) [inline src/selector/input/dynamic-list.ts]: remove this

File: src/selector/input/dynamic-list.ts [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    curr = { ...curr, ...{ [obj.value]: this.getValue(object, obj.key) } },
```
Comment (smurua-co) [inline src/selector/input/dynamic-list.ts]: you don't need to reassign `curr`. Reduce do that automatically
```suggestion
    { ...curr, ...{ [obj.value]: this.getValue(object, obj.key) } },
```

Comment (warseph) [inline src/selector/input/dynamic-list.ts]:
```suggestion
   ( { ...curr, ...{ [obj.value]: this.getValue(object, obj.key) } }),
```

File: src/selector/input/dynamic-list.ts [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+  value: string;
+}
+
+export interface DynamicListInputSelectorSerialized extends Serialized {
+  from: string;
+  path: string;
+  class: 'DynamicListInputSelector';
+}
+
+export class DynamicListInputSelector extends BaseInputSelector
```
Comment (warseph) [inline src/selector/input/dynamic-list.ts]: rename it to something like `DynamicJsonInputSelector`

**Review by warseph — CHANGES_REQUESTED** [review]

**Review by smurua-co — CHANGES_REQUESTED** [review]

File: src/selector/input/dynamic-json.ts [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+  serialize(): DynamicJsonInputSelectorSerialized {
+    return {
+      class: 'DynamicJsonInputSelector',
+      from: this.from,
+      path: JSON.stringify(this.path)
+    };
+  }
+
+  private getValue<T extends {}>(object: T, path: string) {
```
Comment (smurua-co) [inline src/selector/input/dynamic-json.ts:62]: Screen Shot 2019-10-30 at 12 11 40 PM

**Review by warseph — APPROVED** [review]

**Review by smurua-co — APPROVED** [review]

**Pull request merged**